scripts/compile-igs.js

```diff
@@ -0,0 +1,158 @@
+        if (this.options.ignoreVersion) {
+            return name;
+        }
+        return `${name}@${version}`;
+    }
+
+    async extractIgInfo(igDir) {
+        const packagePath = path.join(igDir.toString(), 'package.json');
+        if (!existsSync(packagePath)) {
+            return;
+        }
+        console.log(`checking ${packagePath}`);
+        const packageJson = await loadJson(packagePath);
+        const dependencies = [];
+        const retVal = {
+            id: this.createIGKey(packageJson.name, packageJson.version),
+            url: packageJson.url,
+            name: packageJson.name,
+            version: packageJson.version,
+            dependencies,
+        };
+        const packageDeps = packageJson.dependencies;
+        if (packageDeps) {
+            for (const [name, version] of Object.entries(packageDeps)) {
+                const igId = this.createIGKey(name, version);
+                dependencies.push(igId);
+            }
+        }
+        return retVal;
+    }
+
+    async checkDependencies(igsDir) {
+        const parentMap = {};
+        for (const igPath of await this.listIgDirs(igsDir)) {
+            console.log(`looking at ig path: ${igPath}`);
+            const igInfo = await this.extractIgInfo(igPath);
+            if (igInfo) {
+                parentMap[igInfo.id] = igInfo.dependencies;
+            }
+        }
+        for (const igId of Object.keys(parentMap)) {
+            this.depthFirst([igId], parentMap);
+        }
+    }
+
+    depthFirst(parents, pMap) {
+        const igId = parents[parents.length - 1];
+        const dependencies = pMap[igId];
+        if (!dependencies) {
+            throw new Error(`Missing dependency ${igId}`);
+        }
+        for (const parentId of dependencies) {
+            if (parents.includes(parentId)) {
+                throw new Error(`Circular dependency found: ${parents.join(' -> ')} -> ${parentId}`);
+            }
+            parents.push(parentId);
+            this.depthFirst(parents, pMap);
+            parents.pop();
+        }
+    }
+}
+
+function usage() {
+    console.log('Usage: node compile-igs.js [-i] <directory with IG packs>');
+    console.log('Use -i option to ignore version of dependencies');
+}
+
+/**
+ * main function of the script
+ * parse command line arguments and invoke compiler
+ * * */
+function compileIGs() {
+    const myArgs = process.argv.slice(2);
+    let ignoreVersion = false;
+    let igPath = null;
+    for (const arg of myArgs) {
+        if (arg === '-i') {
+            ignoreVersion = true;
+        } else if (arg === '-h' || arg === 'help') {
+            usage();
+            return;
+        } else {
+            igPath = arg;
+        }
+    }
+    if (igPath === null) {
+        console.error('Need path to IGs folder as argument');
+        usage();
+        process.exit(1);
+    }
+    const options = {
+        ignoreVersion,
+    };
+    return new IGsCompileHelper(options).compileIGs(igPath);
```

Comment:
Errors should be handled here, otherwise you get:
```
 UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
```


It's probably fine to just catch them all and `console.err` them